manifests/katello.pp

```diff
@@ -47,4 +50,40 @@
     alias            => $candlepin_cert_rpm_alias,
     subscribe        => $::certs::server_ca,
   }
+
+  $client_cert_name = "${hostname}-pulp-capsule-client"
+  $client_cert      = "${::certs::pki_dir}/certs/${client_cert_name}.crt"
+  $client_key       = "${::certs::pki_dir}/private/${client_cert_name}.key"
+  $ssl_ca_cert      = $::certs::ca_cert
+
+  # cert for authentication of capsule
+  cert { $client_cert_name:
+    hostname      => $hostname,
+    purpose       => client,
+    country       => $::certs::country,
+    state         => $::certs::state,
+    city          => $::certs::city,
+    org           => 'KATELLO',
+    org_unit      => 'CAPSULE',
+    expiration    => $::certs::expiration,
+    ca            => $::certs::default_ca,
+    generate      => $generate,
+    regenerate    => $regenerate,
+    deploy        => $deploy,
+    password_file => $certs::ca_key_password_file,
+    common_name   => $::pulp::default_login,
```

Comment:
And to clarify that would require a small change to puppet-capsule to pass in the parameter